crates/autopilot/src/run_loop.rs

```diff
@@ -557,7 +548,10 @@ pub fn solve_request(
                     class,
                     app_data: order.data.app_data,
                     signature: order.signature.clone(),
-                    fee_policies,
+                    fee_policies: fee_policies
+                        .get(&order.metadata.uid)
+                        .cloned()
```

Comment:
No need to keep the policies around for orders we already looked at, right?
```suggestion
                        .remove(&order.metadata.uid)
```

Reply:
Added.